c/extras/ftrl.h

```diff
@@ -19,17 +19,23 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 // IN THE SOFTWARE.
 //------------------------------------------------------------------------------
+#ifndef dt_EXTRAS_FTRL_h
+#define dt_EXTRAS_FTRL_h
 #include "py_datatable.h"
+#include "extras/hash.h"
 
-typedef std::unique_ptr<double[]> DoublePtr;
-typedef std::unique_ptr<uint64_t[]> Uint64Ptr;
+
+typedef std::unique_ptr<Hash> hashptr;
+typedef std::unique_ptr<double[]> doubleptr;
+typedef std::unique_ptr<uint64_t[]> uint64ptr;
 #define REPORT_FREQUENCY 1000
 
-struct FtrlModelParams {
-    double a;
-    double b;
-    double l1;
-    double l2;
+
+struct FtrlParams {
+    double alpha;
+    double beta;
+    double lambda1;
+    double lambda2;
     uint64_t d;
```

Comment:
why is this not `size_t` ?

Reply:
Because `uint64_t` is guaranteed to be 64-bit, while `size_t` is not (may be in most of the cases they're identical, but from my understanding there is still a possibility they may differ). Since the Murmur hasher I'm currently using is designed for 64-bits `uint64_t`, I just want to be safe.

Reply:
See `types.cc`: datatable only compiles on platforms where `sizeof(size_t) == sizeof(int64_t) == sizeof(void*) == 8`.

Reply:
Ah, good to know.

Reply:
Guess I will leave this as `uint64_t` for the moment, because otherwise `static_cast` will be needed for each call of Murmur hasher. Will review this in the next PR, may be some adjustments are also needed to the hasher code.